CloverSDK/Utils/CLVEndpoint.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  CLVEndpoint.swift
+//  CloverSDK
+//
+//  Created by Yusuf on 3/9/16.
+//  Copyright © 2016 Clover Network, Inc. All rights reserved.
+//
+
+import Foundation
+
+public enum CLVEndpoint {
+  case V3(CLVV3Endpoint)
+  case CUSTOM(String)
+}
+
+public enum CLVV3Endpoint: String {
+  // Merchants
+  case MERCHANT = "/v3/merchants/{mId}"
+  case MERCHANT_ADDRESS = "/v3/merchants/{mId}/address"
+  case MERCHANT_GATEWAY = "/v3/merchants/{mId}/gateway"
+  case MERCHANT_PROPERTIES = "/v3/merchants/{mId}/properties"
+  case MERCHANT_DEFAULT_SERVICE_CHARGE = "/v3/merchants/{mId}/default_service_charge"
+  case MERCHANT_TIP_SUGGESTIONS = "/v3/merchants/{mId}/tip_suggestions"
+  case MERCHANT_TIP_SUGGESTION = "/v3/merchants/{mId}/tip_suggestions/{tId}"
+  case MERCHANT_ORDER_TYPES = "/v3/merchants/{mId}/order_types"
+  case MERCHANT_ORDER_TYPE = "/v3/merchants/{mId}/order_types/{orderTypeId}"
+  case MERCHANT_ORDER_TYPE_CATEGORIES = "/v3/merchants/{mId}/order_type_categories"
+  case MERCHANT_SYSTEM_ORDER_TYPES = "/v3/merchants/{mId}/system_order_types"
+  case MERCHANT_ROLES = "/v3/merchants/{mId}/roles"
+  case MERCHANT_ROLE = "/v3/merchants/{mId}/roles/{rId}"
+  case MERCHANT_TENDERS = "/v3/merchants/{mId}/tenders"
+  case MERCHANT_TENDER = "/v3/merchants/{mId}/tenders/{tenderId}"
+  case MERCHANT_OPENING_HOURS = "/v3/merchants/{mId}/opening_hours"
+  case MERCHANT_OPENING_HOUR = "/v3/merchants/{mId}/opening_hours/{hId}"
```

Comment:
I actually do have a script to generate code. The CLVMethods file and its PromiseKit extension with methods referring to these endpoints are auto-generated, but with some manual intervention. Puling the endpoints from /v3/api/api_docs is easy-ish, but there's one big problem I couldn't solve: how to connect the endpoint with the returned object type. If I can figure that out, I can modify the script to fully automize that process. On the other hand, we don't change the public endpoints very frequently, so I don't know if that's a necessity.